src/partition/repart.c

```diff
@@ -2908,12 +2908,13 @@ static int context_dump_partitions(Context *context) {
         return table_print_with_pager(t, arg_json_format_flags, arg_pager_flags, arg_legend);
 }
 
-static void context_bar_char_process_partition(
+static int context_bar_char_process_partition(
                 Context *context,
                 Partition *bar[],
                 size_t n,
                 Partition *p,
-                size_t *ret_start) {
+                size_t **ret_start,
+                size_t *n_ret_start) {
```

Comment:
We usually do not use `ret` for arguments updated (instead of assigned) by the function.
Please rename them to e.g. `start_array` and `n_start_array`.